internal/backuprestore/backup.go

```diff
@@ -223,12 +223,20 @@ func (h *BRHandler) cleanupBackupLabels(ctx context.Context, backup *velerov1.Ba
 	if err != nil {
 		return fmt.Errorf("failed to get objs from apply-label annotations: %w", err)
 	}
+
 	// json patch escape info: https://jsonpatch.com/#json-pointer
 	escaped := strings.Replace(backupLabel, "/", "~1", 1)
 	payload := []byte(fmt.Sprintf(`[{"op":"remove","path":"/metadata/labels/%s"}]`, escaped))
+
 	for _, obj := range objs {
-		err := patchObj(ctx, h.DynamicClient, &obj, false, payload) //nolint:gosec
-		if err != nil {
+		patchedObj := obj
+
+		if hasBackupLabel, _ := searchBackupLabel(ctx, h.DynamicClient, &patchedObj); !hasBackupLabel {
```

Comment:
Looks like the deletion is skipped quitely when there is API call error. I think we should add an error log for errors other than "object not found"